src/aliased_struct.h

```diff
@@ -0,0 +1,73 @@
+#ifndef SRC_ALIASED_STRUCT_H_
+#define SRC_ALIASED_STRUCT_H_
+
+#if defined(NODE_WANT_INTERNALS) && NODE_WANT_INTERNALS
+
+#include "node_internals.h"
+#include "v8.h"
+#include <memory>
+
+namespace node {
+
+template <typename T,
+          typename = std::enable_if_t<std::is_object<T>::value>>
+class AliasedStruct {
+ public:
+  explicit AliasedStruct(v8::Isolate* isolate) : isolate_(isolate) {
+    const v8::HandleScope handle_scope(isolate);
+
+    store_ = v8::ArrayBuffer::NewBackingStore(isolate, sizeof(T));
+    ptr_ = new (store_->Data()) T;
+    DCHECK_NOT_NULL(ptr_);
+
+    v8::Local<v8::ArrayBuffer> buffer = v8::ArrayBuffer::New(isolate, store_);
+    buffer_ = v8::Global<v8::ArrayBuffer>(isolate, buffer);
+  }
+
+  AliasedStruct(const AliasedStruct& that)
```

Comment:
Is it intended that the copy constructor doesn't copy the data just create one more reference?
Isn't it a problem that two instances point to  the same T which may be deleted twice then?

Reply:
Fwiw, with the suggestion above applied, we could solve that by turning this into 

```c++
  AliasedStruct(const AliasedStruct& that) : AliasedStruct(that.isolate_, *that) {}
```

Reply:
@addaleax ... just want to confirm.. with which suggestion? [This one](https://github.com/nodejs/node/pull/32778#discussion_r407062287)?

Reply:
Another option would be to remove/delelet the copy construtor as it seems to be not needed anyway. But it's fine as is is also.

Reply:
@jasnell Yeah, that was the one I was talking about :+1: